### ISSUE TITLE
Trouble finding anything other than movies

### ISSUE THREAD

**Issue by digiltd** [kodi-connect/kodi-alexa-video#1]
Hi

I thought I would start the conversation going with an issue.

I am having real difficulty getting it to play TV shows.

"Alexa, play the big bang theory"

"Alexa, play an episode of big bang theory"

"Alexa, play the tv show the big bang theory"

Nothing seems to work. As you have been working with the Video Skill Kit, can you advise on what she is listening out for?

**Comment by Mahalo1218** [comment]
It works just fine. I ask 'Alexa play the big bang theory' it starts with
the first Episode. You can also specify what season and episode you want. I
have only had trouble with some movies.

**Comment by digiltd** [comment]
Strange. Here is the log

![mac mini 2018-01-31 15-13-34](https://user-images.githubusercontent.com/5680968/35630554-a1e750ec-0699-11e8-89d0-0dd9de24fa69.png)

fyi, I have none of these other than the actual show... especially not "Dennis Rodman's Big Bang in Pyongyang" 😄

**Comment by Mahalo1218** [comment]
I don't know

**Comment by tkislan** [comment]
And do you have everything in your Video Library? Because Kodi Addon doesn't look through filesystem, it only queries Kodi's Video Library.

Also, by default it plays next unwatched episode. So if you've watched everything, it won't play anything. And as @Mahalo1218 mentioned, you can try specifying both season and episode, to watch an already watched episode.

**Comment by digiltd** [comment]
It is all in the library, local files in folders, all scraped with the usual kodi way and accessible through the jsonrpc. 

It might be because I have seen them all, yep just tested it by marking one as unwatched, and it worked.

I did get a "something went wrong" message, but it played it anyway. Tested this three times and it always seems to happen at the exact moment it starts to play. What should it say? I am aware of the Alexa timeout issue, but this seems too consistent.

It takes 9-11 seconds to start playing. This happens each time. Which does seem a long time as the kodi-alexa skill takes 2-3 seconds. I wonder if it is searching for each of the generated "guesses" shown in the log above? Do you fuzzymatch them?

**Comment by tkislan** [comment]
@digiltd Yes, it's doing fuzzymatch across whole library .. which might be pretty slow on some devices, but was working quite fast on my RPi3

I'm having similar problem, because I'm using external HDD, and it takes some time to spin it up to start playback.

I already have new version of Addon ready, that will lookup stuff in async call, so it responds to Alexa as soon as it receives the message, and then display notification on screen, if it doesn't find anything, or any other, more helpful info.

**Comment by digiltd** [comment]
Cool

Maybe the fuzzymatching needs a bit of tweaking? 

I am running kodi on a macmini (late 2012 2.5GHz dual-core Intel Core i5) with 16gb of ram. Whilst I know it is not a beast by today's standards, in terms of a htpc it is still pretty beefy and shouldn't take 10 seconds for the search. I keep it clean and only have 57 shows 166 movies, and the disc has already spun up when I tried the commands. I will have a look as well.

**Comment by tkislan** [comment]
When testing on my Macbook Pro, it takes around 200ms to search through all actors in my library, which is around 5000 entries.
So there is definitely something else, that's taking so long

**Comment by digiltd** [comment]
Interesting, what command are you using and is the requesting going through the skill or testing locally?

**Comment by tkislan** [comment]
You can check multiple unit tests for the Addon, to see how it's run locally.

**Comment by m1dy** [comment]
For large libraries searches are very slow for me. 
Tested 2 Kodi-instances on a i5 2500k @4.4ghz with different libraries:
~300 movies ~100 TV Shows: one request took about 2-5 seconds (which is fine)
~9000 movies ~1800 TV Shows: one request took >2 minutes and even when it started playing cpu load was still 100% (some bug?)
Just commented in case it helps somehow...

**Comment by tkislan** [comment]
@m1dy Wow, that's huge. Thanks for insights
Yeah I've had a trigram indexing done to make it crazy fast, but it's dependent on numpy, which is a binary module, and can't be distributed with the addon

Will have to come up with some indexing in pure python

**Comment by tkislan** [comment]
@m1dy New version is available, that implements trigram indexing, so the searches are super fast, even though it takes some time to build the index itself

Please try it out, and let me know if it helped
https://www.dropbox.com/s/hkgs1d03ueep428/plugin.video.kodiconnect-0.1.5.zip?dl=0

**Comment by m1dy** [comment]
@tkislan wow... now it finds movies/tvshows instantly even with my large library
nice update!
but when exactly does the plugin decide to index? On start of kodi? when scan is finished?

**Comment by tkislan** [comment]
Yes and yes. It does the initial scan after around 20 sec of starting up (this I have to tweak a bit)
and it should update library after every finished scan .. but this I have to test first, because last time it didn't pick up new movies for some reason

**Comment by tkislan** [comment]
@digiltd Can you please recheck with the new version of addon? So that I can close this issue

**Comment by digiltd** [comment]
it seems to find it fast enough, thanks

"show me the big bang theory" showed me a list of results: The Big Bang Theory, The BFG, The Lion King, The Jungle Book, The Big Family Cooking Showdown. Though maybe that is another issue :)

Is there anyway to say "there are no new episodes of The Big Bang Theory" when it can't find any unwatched episodes, or is that something Amazon set?

Cheers

**Comment by tkislan** [comment]
Threshold for title is currently kinda low, which is not a problem for playing, since it only takes the best match.
Will look into the showing part :)

Unfortunately no, there is a fixed interface.
Currently I'm thinking between showing a notification, that there are no new episodes, or just playing a random episode from that show
Or even better, have this as an option in addon settings
